### PR TITLE
Add further tests of cyclical project imports

### PR DESCRIPTION
**Pull request by philderbeast**
While reviewing #9578, I saw that the cyclical test didn't `assertOutputContains` and had but one cyclical test, a simple loopback of a project importing itself. It now has the following test projects;

```
+-- cyclical-0-self.project (imports cyclical-0-self.project)
+-- cyclical-0-self.project (already processed)
+-- etc
```

```
+-- cyclical-1-out-back.project
 +-- cyclical-1-out-back.config (imports cyclical-1-out-back.project)
+-- cyclical-1-out-back.project (already processed)
 +-- etc
```

```
+-- cyclical-1-out-self.project
 +-- cyclical-1-out-self.config (imports cyclical-1-out-self.config)
 +-- cyclical-1-out-self.config (already processed)
 +-- etc
```

```
+-- cyclical-2-out-out-backback.project
 +-- cyclical-2-out-out-backback-a.config
  +-- cyclical-2-out-out-backback-b.config (imports cyclical-2-out-out-backback.project)
+-- cyclical-2-out-out-backback.project (already processed)
 +-- etc
 ```
 
 ```
+-- cyclical-2-out-out-back.project
 +-- cyclical-2-out-out-back-a.config
  +-- cyclical-2-out-out-back-b.config (imports cyclical-2-out-out-back-a.config)
 +-- cyclical-2-out-out-back-a.config (already processed)
  +-- etc
```

```
+-- cyclical-2-out-out-self.project
 +-- cyclical-2-out-out-self-a.config
  +-- cyclical-2-out-out-self-b.config (imports cyclical-2-out-out-self-b.config)
  +-- cyclical-2-out-out-self-b.config (already processed)
  +-- etc
```

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).

### CODE REVIEW

File: cabal-testsuite/PackageTests/ConditionalAndImport/cyclical-2-out-out-back-a.config
Comment (geekosaur) [inline cabal-testsuite/PackageTests/ConditionalAndImport/cyclical-2-out-out-back-a.config:1]: I think this file is missing a newline at the end?

File: cabal-testsuite/PackageTests/ConditionalAndImport/cyclical-2-out-out-back-b.config
Comment (geekosaur) [inline cabal-testsuite/PackageTests/ConditionalAndImport/cyclical-2-out-out-back-b.config:1]: Same here.

File: cabal-testsuite/PackageTests/ConditionalAndImport/cyclical-2-out-out-backback-a.config
Comment (geekosaur) [inline cabal-testsuite/PackageTests/ConditionalAndImport/cyclical-2-out-out-backback-a.config:1]: And here.

File: cabal-testsuite/PackageTests/ConditionalAndImport/cyclical-2-out-out-backback-b.config
Comment (geekosaur) [inline cabal-testsuite/PackageTests/ConditionalAndImport/cyclical-2-out-out-backback-b.config:1]: And here.

File: cabal-testsuite/PackageTests/ConditionalAndImport/cyclical-2-out-out-self-a.config
Comment (geekosaur) [inline cabal-testsuite/PackageTests/ConditionalAndImport/cyclical-2-out-out-self-a.config:1]: And here.

File: cabal-testsuite/PackageTests/ConditionalAndImport/cyclical-2-out-out-self-b.config
Comment (geekosaur) [inline cabal-testsuite/PackageTests/ConditionalAndImport/cyclical-2-out-out-self-b.config:1]: And here.

**Review by geekosaur — CHANGES_REQUESTED** [review]

File: cabal-testsuite/PackageTests/ConditionalAndImport/cyclical-2-out-out-back-a.config
Comment (philderbeast) [reply cabal-testsuite/PackageTests/ConditionalAndImport/cyclical-2-out-out-back-a.config:1]: Easy to miss, sorry.

Comment (philderbeast) [reply cabal-testsuite/PackageTests/ConditionalAndImport/cyclical-2-out-out-back-a.config:1]: Done.

**Comment by philderbeast** [comment]
Thanks @geekosaur, I've added newlines where they were missing in all files.

**Review by geekosaur — APPROVED** [review]

**Comment by philderbeast** [comment]
@Mikolaj if this gets approval, could we waive the 2 day delay please? It expands tests, tests that I help with #9578.

**Comment by philderbeast** [comment]
Thanks @Mikolaj. I've squashed down to two commits, one that adds `assertOutputContains` to make the existing test more robust and another for adding the additional tests.

**Comment by philderbeast** [comment]
Label merge+no rebase is necessary when the pull request is from an organisation.

**Pull request merged**